documentation/source/newsfragments/3354.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fix incorrect cleanup of pending Tasks (queued by :func:`cocotb.start_soon` but not started yet) when a test ends.
```

Comment:
Can you directly edit the release_notes.rst? Towncrier doesn't rerun on branch releases.